### PR TITLE
Feature/linear fp

### PR DESCRIPTION
**Pull request by hroest**
Add a small function for outside clients to compute a linear fixed point. Instead of computing the maximal possible fixpoint, this computes the minimal fixpoint to achieve the desired mass accuracy. 

Using this function results in substantially smaller file sizes. I observe 70% smaller files in some instances by simply changing this parameter from the maximal accuracy to ca 1 ppm accuracy @ 150 mass-to-charge (which is about 0.5 ppm accuracy at 300 m/z). This is more than sufficient for most TOF data and brings down the file sizes substantially:

As a test file I use the file `olgas_K121026_001_SW_Wayne_R1_d00.wiff` from the Schubert et al [1,2]:

```
-rw-rw-r-- 1 hr   hr   8.6M Dec  6  2015 olgas_K121026_001_SW_Wayne_R1_d00.wiff
-rw-rw-r-- 1 hr   hr   3.5G Dec  6  2015 olgas_K121026_001_SW_Wayne_R1_d00.wiff.scan
```

when using a regular converter to mzML, I end up with a file of about 15 GB size in profile mode (no numpress compression but `.gz` on the file). Applying the currently implemented conversion in proteowizard / OpenMS with the "maximal" fixed point results in a file of 5.6 GB size (using `.gz`) but when specifying 1ppm accuracy at either 150 m/z or 400 m/z, the file size goes down to 3.1 to 3.3 GB:

```
-rw-r--r-- 1 hr hr    15G Mar  3 22:28 olgas_K121026_001_SW_Wayne_R1_d00.wiff.profile.mzML.gz
-rw-rw-r-- 1 hr hr   5.6G Mar  7 18:37 olgas_K121026_001_SW_Wayne_R1_d00.wiff.profile.compress.mzML.gz
-rw-rw-r-- 1 hr hr   3.3G Mar  8 14:03 olgas_K121026_001_SW_Wayne_R1_d00.wiff.profile.lc_1ppm_150.mzML.gz
-rw-rw-r-- 1 hr hr   3.1G Mar  8 13:02 olgas_K121026_001_SW_Wayne_R1_d00.wiff.profile.lc_1ppm_400.mzML.gz
```

Including this function allows client to request a fixed point that minimizes file size while at the same time preserving a desired accuracy. While the computation is trivial, the interface makes it easier for clients to get a guarantee for the specific accuracy and the client does not have to understand how the fixed Point is calculated. 

Basically, this PR will allow clients to reduce their file sizes by a large amount by choosing appropriate parameters.

Ref

1. http://www.biorxiv.org/content/early/2016/03/18/044552
2. http://www.peptideatlas.org/PASS/PASS00779

### CODE REVIEW

File: src/main/cpp/MSNumpressTest.cpp [hunk 1]
```diff
@@ -109,6 +108,87 @@ void decodeLinearNice() {
...
+  
+  double mz_err[7];
+  double encodedLength[7];
+
+  // for higher accuracy, we get longer encoded lengths
+  mz_err[0] = 0.1;  encodedLength[0] = 22;
+  mz_err[1] = 1e-3; encodedLength[1] = 25;
+  mz_err[2] = 1e-5; encodedLength[2] = 29;
+  mz_err[3] = 1e-6; encodedLength[3] = 30;
+  mz_err[3] = 1e-7; encodedLength[3] = 31;
```
Comment (hroest) [inline src/main/cpp/MSNumpressTest.cpp]: this gives an illustration how the length is dependent on the desired accuracy

**Comment by hroest** [comment]
ok, I added some tests for C and Python. As you can see, this also tests how the encoded length depends on the fixed point and the desired accuracy

File: src/main/cpp/MSNumpressTest.cpp (hunk 1 above)
Comment (fickludd) [inline src/main/cpp/MSNumpressTest.cpp]: Good! I think you have overwrite the `ms_err` and `encodedLength` of L168 though...

**Comment by fickludd** [comment]
I think you could make a small technical brief on this if you feel like a low-hanging paper. Just grab a bunch of mzML files, compress with increasing error tolerance, and see how this effects the analytical results from some different tools. One figure, two pages. Up for grabs =). Maybe @flevander wants to collaborate?  

Also, I though you were doing genomics nowadays?

**Comment by hroest** [comment]
Sounds interesting, I hope of course it wont affect the results from different tools too much - e.g. for SWATH analysis we bin most of our data at 30 ppm so having an accuracy loss of < 1ppm should not make a big difference. I would be happy to collaborate -- But these are just minor additions in how to choose the parameters in the algorithm that you have written so credit belongs to you. I still think its nice to get to the same file size as .wiff files. Maybe they use something similar internally?

Yes, I am doing many things these days, also some genomics but still a lot of mass spectrometry.

**Comment by fickludd** [comment]
Well no, the results should not differ if you compress at 1ppm and later bin at 30ppm. But how far can you go? What if you compress at 2ppm? 5ppm? 30? What you can get half the `.wiff` file size without loss of analytical results in the next step?

I published the algorithm, so it's free for anyone to use and augment, as you have just done =). I actually don't think the "algorithm" invention would be such a big part of that paper, but rather taking the time to gather files, run the compressions and all the tools, and then do the whole paper thing. Whoever does that deserves the credit.

You got too good at MS and now it won't let you go =). If you would be interested in switching over to industry, we are hiring btw (https://neo4j.com/jobs/).
